lib/sycamore/sycamore/query/operators/query_database.py

```diff
@@ -42,3 +42,54 @@ class QueryDatabase(LogicalOperator):
     Whenever possible, use the query parameter to filter data at the source, as this is more
     efficient than filtering data in subsequent data filtering operators.
     """
+
+
+class QueryVectorDatabase(LogicalOperator):
+    """OpenSearch knn query to load records that are similar to a given query string. Returns the top k records
+    according to the cosine similarity between the query string and record's text content.
+    Use this if we need to filter on a field that can't suffice with an exact match, but also isn't complex enough to
+    require a llm_filter, i.e. a record level LLM call.
+
+    Here are some instructions about using QueryVectorDatabase instead of QueryDatabase:
+    1. Use QueryVectorDatabase if the query plan uses an LLM operator (LLMFilter or LLMExtractEntity),
+        and the final answer doesn't require you to scan the full dataset (e.g. find me an instance of a transcript
+        that talked about AI)
+    2. Use QueryVectorDatabase for RAG-style questions where you need to retrieve a set of candidate records where the
+        answer might exist.
+    3. Do not use QueryVectorDatabase where examining each record is essential (e.g. did all transcripts talk about AI?)
```

Comment:
Can you make sure each of these list items ends in a period? This is SUPER anal, I know, but I have a superstitious belief that grammatical queries do a better job than ones with missing punctuation, etc. (This is a completely unfounded belief, but it seems to me that being grammatical is always better when it comes to prompting LLMs.)

Reply:
Haha sure